src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -7470,7 +7470,8 @@ private BoundExpression BindIndexedPropertyAccess(SyntaxNode syntax, BoundExpres
             // to this indexer that has an Index or Range type and that there is
             // a real receiver with a known type
 
-            if (arguments.Count != 1)
+            if (!Compilation.IsFeatureEnabled(MessageID.IDS_FeatureIndexOperator) ||
```

Comment:
>if (!Compilation.IsFeatureEnabled(MessageID.IDS_FeatureIndexOperator) || [](start = 12, length = 72)

This feels wrong. We should always bind as best we can (ie. as-if all features were enabled). The only difference should be the diagnostics produced.

Reply:
Ah, that's a good point. I will change this.